### PR TITLE
Schema: Remove `key` from root DataQuery type

### PR DESCRIPTION
**Pull request by ryantxu**
This removes `key` from DataQuery schema -- it is a property that exists and is used only in explore mode.

In general `refId` is used to identify different queries

### CODE REVIEW

File: pkg/kindsys/common_dataquery.cue [hunk 1]
```diff
@@ -9,15 +9,16 @@ package kindsys
...
+	// By default, the UI will assign A-Z; however setting meaninful names may be useful
 	refId: string
 
 	// true if query is disabled (ie should not be returned to the dashboard)
+	// Note this does not always imply that the query should not be executed since
+	// the results from a hidden query may be used as the input to other queries (SSE etc)
 	hide?: bool
 
-	// Unique, guid like, string used in explore mode
-	key?: string
```
Comment (ryantxu) [inline pkg/kindsys/common_dataquery.cue:19]: ^^^ removed here 

(updating comments in this file... everything else is generated)

File: packages/grafana-schema/src/veneer/common.types.ts [hunk 2]
```diff
@@ -9,6 +9,11 @@ export interface MapLayerOptions<TConfig = any> extends raw.MapLayerOptions {
 }
 
 export interface DataQuery extends raw.DataQuery {
+  /**
+   * Unique, guid like, string (used only in explore mode)
+   */
+  key?: string;
```
Comment (ryantxu) [inline packages/grafana-schema/src/veneer/common.types.ts:15]: ^^^ added explicitly to the typescript value here

**Review by sdboyer — APPROVED** [review]
thank you! the costs of doing this stuff blind to its actual use 😧

File: pkg/kindsys/common_dataquery.cue [hunk 3]
```diff
@@ -9,15 +9,16 @@ package kindsys
...
-
 	// Specify the query flavor
 	// TODO make this required and give it a default
 	queryType?: string
```
[the remark this answers is not in the document]
Comment (sdboyer) [reply pkg/kindsys/common_dataquery.cue:24]: this one is really nasty, tbh - it's why #60680 is still open. It's now clear to me that writing and codegen-ing dataquery compo kinds is made less clear by the fact that we don't build in to the schema interface definition the notion that this is a discriminator field...when the field exists.

To actually make that change, IMO we need a design doc specifically for the `DataQuery` schema interface to make sure we're covering all the angles

[the remark this answers is not in the document]
Comment (ryantxu) [reply pkg/kindsys/common_dataquery.cue:24]: yup -- there are really two modes used here.  In some cases, `queryType` changes the set of valid values and in others they just all live as siblings.

The github datasource is a good example of the queryType dependant models:
https://github.com/grafana/github-datasource/blob/main/pkg/models/query.go#L42

**Pull request merged**